src/charm.py

```diff
@@ -18,6 +23,82 @@ class SpringBootCharm(CharmBase):
     def __init__(self, *args):
         """Initialize the instance."""
         super().__init__(*args)
+        self.framework.observe(self.on.config_changed, self.reconciliation)
+
+    def _detect_java_application(self) -> ExecutableJarApplication:
+        """Detect the type of the Java application inside the Spring Boot application image."""
+        container = self._spring_boot_container()
+        if container.isdir("/app"):
```

Comment:
Do you think would be worth creating a constant for "/app"?

Reply:
This will later be replaced by a charm config, so for the first iteration that's fine for me